sharding/partitions.go

```diff
@@ -233,19 +237,22 @@ func (p *Partitions) updateRemote(nodes []string) {
 }
 
 func (p *Partitions) updateMissing() {
-	// Check for each partition. If every one is available on at least one node,
+	// Check for each partition. If every one is available on at least minReplication node,
 	// then we're ready to rumble.
 	missing := 0
 	for i := 0; i < p.numPartitions; i++ {
+		replication := 0
 		if _, ok := p.local[i]; ok {
-			continue
+			replication++
 		}
 
-		if _, ok := p.remote[i]; ok {
-			continue
+		if remotes, ok := p.remote[i]; ok {
+			replication += len(remotes)
 		}
 
-		missing += 1
+		if replication < p.minReplication {
+			missing += 1
+		}
 	}
 
 	p.numMissing = missing
```

Comment:
I think we could actually delete `p.numMissing`? The only thing that reads it is `Missing()`, which isn't used anywhere.